cmd/processor/main.go

```diff
@@ -0,0 +1,208 @@
+)
+
+type processorConfig struct {
+	LogLevel       string
+	LeaderElection bool
+	PodNamespace   string
+	LeaseDuration  time.Duration
+	RenewDeadline  time.Duration
+	RetryPeriod    time.Duration
+}
+
+func parseEnvFlags() processorConfig {
+	viper.SetDefault(logLevelFlag, "Info")
+	viper.SetDefault(leaderElectionFlag, false)
+	viper.SetDefault(podNamespace, "")
+	viper.SetDefault(leaseDurationFlag, 15*time.Second)
+	viper.SetDefault(renewDeadlineFlag, 10*time.Second)
+	viper.SetDefault(retryPeriodFlag, 2*time.Second)
+
+	pflag.String(logLevelFlag, viper.GetString(logLevelFlag), "Log level")
+	pflag.Bool(leaderElectionFlag, viper.GetBool(leaderElectionFlag), "Enable leader election")
+	pflag.String(podNamespace, viper.GetString(podNamespace), "Pod namespace")
+	pflag.Duration(leaseDurationFlag, viper.GetDuration(leaseDurationFlag), "Leader election lease duration")
+	pflag.Duration(renewDeadlineFlag, viper.GetDuration(renewDeadlineFlag), "Leader election renew deadline")
+	pflag.Duration(retryPeriodFlag, viper.GetDuration(retryPeriodFlag), "Leader election retry period")
+	pflag.Parse()
+
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+	viper.AutomaticEnv()
+	_ = viper.BindPFlags(pflag.CommandLine)
+
+	return processorConfig{
+		LogLevel:       viper.GetString(logLevelFlag),
+		LeaderElection: viper.GetBool(leaderElectionFlag),
+		PodNamespace:   viper.GetString(podNamespace),
+		LeaseDuration:  viper.GetDuration(leaseDurationFlag),
+		RenewDeadline:  viper.GetDuration(renewDeadlineFlag),
+		RetryPeriod:    viper.GetDuration(retryPeriodFlag),
+	}
+}
+
+func main() {
+	ctx, cancelCtx := context.WithCancel(context.Background())
+	defer cancelCtx()
+
+	conf := parseEnvFlags()
+
+	logger.WithField("version", pkg.Version).WithField("processorConf", conf).
+		WithField("featureGates", runtime.EncodeFeatures()).
+		Info("Starting agones-processor")
+
+	logger.WithField("logLevel", conf.LogLevel).Info("Setting LogLevel configuration")
+	level, err := logrus.ParseLevel(strings.ToLower(conf.LogLevel))
+	if err == nil {
+		runtime.SetLevel(level)
+	} else {
+		logger.WithError(err).Info("Specified wrong Logging. Setting default loglevel - Info")
+		runtime.SetLevel(logrus.InfoLevel)
+	}
+
+	healthserver := &httpserver.Server{Logger: logger}
+	health := healthcheck.NewHandler()
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logger.WithError(err).Fatal("Failed to create in-cluster config")
+		panic("Failed to create in-cluster config: " + err.Error())
+	}
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logger.WithError(err).Fatal("Failed to create Kubernetes client")
+		panic("Failed to create Kubernetes client: " + err.Error())
+	}
+
+	go func() {
+		healthserver.Handle("/", health)
+		_ = healthserver.Run(context.Background(), 0)
+	}()
+
+	signals.NewSigTermHandler(func() {
+		logger.Info("Pod shutdown has been requested, failing readiness check")
+		cancelCtx()
+		time.Sleep(1 * time.Second)
```

Comment:
Do we need a minute sleep timer before shutdown? Is this to handle any remaining processing items that are mid flight? (do we care? maybe we do).

If so, should it be configurable like it is in the allocator?

Reply:
I should probably get rid of it from this PR as it's just the scaffolding, it will indeed be configurable on the implementation of the allocator on it 😄

Reply:
Updated: [a9b200f](https://github.com/googleforgames/agones/pull/4222/commits/a9b200f583e14c0fc423b3c2b438e49de6c07e17)